torchaudio/prototype/tacotron2.py

```diff
@@ -0,0 +1,847 @@
+        attention_hidden_dim (int): dimension of attention hidden representation
+        attention_location_n_filter (int): number of filters for attention model
+        attention_location_kernel_size (int): kernel size for attention model
+        attention_dropout (float): dropout probability for attention LSTM
+        prenet_dim (int): number of ReLU units in prenet layers
+        gate_threshold (float): probability threshold for stop token
+    """
+
+    def __init__(self,
+                 n_mel: int,
+                 n_frames_per_step: int,
+                 encoder_embedding_dim: int,
+                 decoder_rnn_dim: int,
+                 decoder_max_step: int,
+                 decoder_dropout: float,
+                 decoder_early_stopping: bool,
+                 attention_rnn_dim: int,
+                 attention_hidden_dim: int,
+                 attention_location_n_filter: int,
+                 attention_location_kernel_size: int,
+                 attention_dropout: float,
+                 prenet_dim: int,
+                 gate_threshold: float) -> None:
+
+        super().__init__()
+        self.n_mel = n_mel
+        self.n_frames_per_step = n_frames_per_step
+        self.encoder_embedding_dim = encoder_embedding_dim
+        self.attention_rnn_dim = attention_rnn_dim
+        self.decoder_rnn_dim = decoder_rnn_dim
+        self.prenet_dim = prenet_dim
+        self.decoder_max_step = decoder_max_step
+        self.gate_threshold = gate_threshold
+        self.attention_dropout = attention_dropout
+        self.decoder_dropout = decoder_dropout
+        self.decoder_early_stopping = decoder_early_stopping
+
+        self.prenet = _Prenet(
+            n_mel * n_frames_per_step,
+            [prenet_dim, prenet_dim])
+
+        self.attention_rnn = nn.LSTMCell(
+            prenet_dim + encoder_embedding_dim,
+            attention_rnn_dim)
+
+        self.attention_layer = _Attention(
+            attention_rnn_dim, encoder_embedding_dim,
+            attention_hidden_dim, attention_location_n_filter,
+            attention_location_kernel_size)
+
+        self.decoder_rnn = nn.LSTMCell(
+            attention_rnn_dim + encoder_embedding_dim,
+            decoder_rnn_dim, True)
+
+        self.linear_projection = _LinearNorm(
+            decoder_rnn_dim + encoder_embedding_dim,
+            n_mel * n_frames_per_step)
+
+        self.gate_layer = _LinearNorm(
+            decoder_rnn_dim + encoder_embedding_dim, 1,
+            bias=True, w_init_gain='sigmoid')
+
+    def _get_initial_frame(self, memory: Tensor) -> Tensor:
+        r"""Gets all zeros frames to use as the first decoder input.
+
+        Args:
+            memory (Tensor): Encoder outputs (n_batch, text_lengths.max(), encoder_embedding_dim)
+
+        Returns:
+            decoder_input (Tensor): all zeros frames (n_batch, text_lengths.max(), n_mel * n_frames_per_step)
+        """
+
+        B = memory.size(0)
```

Comment:
Can we replace `B` with something more readable? something like `batch_size`.

Reply:
Thanks for pointing it out. It's addressed [here](https://github.com/pytorch/audio/pull/1621/commits/68b127377f8f8f8b6ae64c06ff4f6de885335d57#diff-8b8ec57192b9aaeb99a9ec5e709c977771b31b49fe7ddc6704c2128cf161a00eR501).